File: flutter_app/lib/screens/snooze_options_screen.dart
import 'package:flutter/material.dart';
import 'package:go_router/go_router.dart';
import 'package:provider/provider.dart';
import '../providers/app_state.dart';
import '../theme/app_colors.dart';
import '../widgets/care_header.dart';

class SnoozeOptionsScreen extends StatefulWidget {
  const SnoozeOptionsScreen({super.key});

  @override
  State<SnoozeOptionsScreen> createState() => _SnoozeOptionsScreenState();
}

class _SnoozeOptionsScreenState extends State<SnoozeOptionsScreen> {
  String _selected = '15';

  static const _options = [
    {'value': '5', 'label': '5 Minutes', 'desc': 'Quick pause'},
    {'value': '15', 'label': '15 Minutes', 'desc': 'Short break'},
    {'value': '30', 'label': '30 Minutes', 'desc': 'Take your time'},
    {'value': '60', 'label': '1 Hour', 'desc': 'Longer delay'},
    {'value': '120', 'label': '2 Hours', 'desc': 'Extended snooze'},
  ];

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.pageBg,
      body: Column(
        children: [
          CareHeader(
            title: 'Snooze Reminder',
            onBack: () => context.pop(),
          ),
          Expanded(
            child: SingleChildScrollView(
              padding: const EdgeInsets.all(24),
              child: Column(
                crossAxisAlignment: CrossAxisAlignment.start,
                children: [
                  const Text(
                    'How long would you like to wait?',
                    style: TextStyle(
                      fontSize: 24,
                      fontWeight: FontWeight.w900,
                      color: AppColors.heading,
                    ),
                  ),
                  const SizedBox(height: 8),
                  const Text(
                    'Choose when you would like to be reminded again.',
                    style: TextStyle(
                      fontSize: 16,
                      fontWeight: FontWeight.w500,
                      color: AppColors.mutedText,
                    ),
                  ),
                  const SizedBox(height: 24),
                  ..._options.map((opt) => _buildOption(opt)),
                ],
              ),
            ),
          ),
          _buildFooter(),
        ],
      ),
    );
  }

  Widget _buildOption(Map<String, String> opt) {
    final isSelected = _selected == opt['value'];
    return Padding(
      padding: const EdgeInsets.only(bottom: 12),
      child: GestureDetector(
        onTap: () => setState(() => _selected = opt['value']!),
        child: Container(
          padding: const EdgeInsets.all(20),
          decoration: BoxDecoration(
            color: isSelected ? AppColors.blueBg : AppColors.white,
            borderRadius: BorderRadius.circular(24),
            border: Border.all(
              color: isSelected ? AppColors.primaryAction : AppColors.border,
              width: 4,
            ),
          ),
          child: Row(
            children: [
              Container(
                padding: const EdgeInsets.all(10),
                decoration: BoxDecoration(
                  color: isSelected ? AppColors.primaryAction : AppColors.subtleBg,
                  borderRadius: BorderRadius.circular(14),
                ),
                child: Icon(
                  Icons.snooze,
                  size: 24,
                  color: isSelected ? AppColors.white : AppColors.mutedText,
                ),
              ),
              const SizedBox(width: 16),
              Expanded(
                child: Column(
                  crossAxisAlignment: CrossAxisAlignment.start,
                  children: [
                    Text(
                      opt['label']!,
                      style: TextStyle(
                        fontSize: 20,
                        fontWeight: FontWeight.w700,
                        color: isSelected ? AppColors.primaryAction : AppColors.heading,
                      ),
                    ),
                    Text(
                      opt['desc']!,
                      style: const TextStyle(fontSize: 16, color: AppColors.mutedText),
                    ),
                  ],
                ),
              ),
              if (isSelected)
                const Icon(Icons.check_circle, color: AppColors.primaryAction, size: 28),
            ],
          ),
        ),
      ),
    );
  }

  Widget _buildFooter() {
    return Container(
      padding: const EdgeInsets.all(24),
      decoration: const BoxDecoration(
        color: AppColors.white,
        border: Border(top: BorderSide(color: AppColors.border, width: 4)),
      ),
      child: SafeArea(
        top: false,
        child: SizedBox(
          width: double.infinity,
          height: 64,
          child: ElevatedButton(
            onPressed: () {
              final appState = context.read<AppState>();
              if (appState.reminders.isNotEmpty) {
                appState.snoozeReminder(appState.reminders.first.id);
              }
              context.go('/success?type=snooze&title=Medication');
            },
            style: ElevatedButton.styleFrom(
              backgroundColor: AppColors.primaryAction,
              foregroundColor: AppColors.white,
              shape: RoundedRectangleBorder(
                borderRadius: BorderRadius.circular(20),
              ),
              textStyle: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700),
            ),
            child: const Text('Snooze Reminder'),
          ),
        ),
      ),
    );
  }
}

File: react-native-app/src/screens/SnoozeOptionsScreen.js
import React, { useState } from 'react';
import { View, Text, TouchableOpacity, ScrollView, StyleSheet } from 'react-native';
import { SafeAreaView } from 'react-native-safe-area-context';
import { Ionicons } from '@expo/vector-icons';
import CareHeader from '../components/CareHeader';
import Colors from '../theme/colors';
import { useAppState } from '../context/AppContext';

const DURATIONS = ['5 minutes', '15 minutes', '30 minutes', '1 hour', '2 hours'];

export default function SnoozeOptionsScreen({ navigation }) {
  const { state, snoozeReminder } = useAppState();
  const [selected, setSelected] = useState('15 minutes');
  const reminder = state.reminders.length > 0 ? state.reminders[0] : null;

  const handleSnooze = () => {
    if (reminder) snoozeReminder(reminder.id);
    navigation.navigate('Success', { type: 'snooze', title: 'Medication' });
  };

  return (
    <SafeAreaView style={styles.safe}>
      <CareHeader title="Snooze Reminder" onBack={() => navigation.goBack()} />
      <ScrollView contentContainerStyle={styles.scroll}>
        <Text style={styles.heading}>How long would you like to snooze?</Text>
        {DURATIONS.map((d) => (
          <TouchableOpacity
            key={d}
            style={[styles.option, selected === d && styles.optionSelected]}
            onPress={() => setSelected(d)}
            accessibilityRole="radio"
            accessibilityState={{ selected: selected === d }}
          >
            <Ionicons
              name={selected === d ? 'radio-button-on' : 'radio-button-off'}
              size={24}
              color={selected === d ? Colors.primaryAction : Colors.disabledText}
            />
            <Text style={[styles.optionText, selected === d && styles.optionTextSelected]}>{d}</Text>
          </TouchableOpacity>
        ))}
        <View style={{ height: 24 }} />
        <TouchableOpacity style={styles.primaryButton} onPress={handleSnooze}>
          <Text style={styles.primaryText}>Snooze Reminder</Text>
        </TouchableOpacity>
      </ScrollView>
    </SafeAreaView>
  );
}

const styles = StyleSheet.create({
  safe: { flex: 1, backgroundColor: Colors.pageBg },
  scroll: { padding: 24 },
  heading: { fontSize: 22, fontWeight: '900', color: Colors.heading, marginBottom: 24 },
  option: {
    flexDirection: 'row', alignItems: 'center', gap: 16,
    padding: 20, borderRadius: 20, marginBottom: 12,
    backgroundColor: Colors.white, borderWidth: 3, borderColor: Colors.border,
  },
  optionSelected: { borderColor: Colors.primaryAction, backgroundColor: Colors.blueBg },
  optionText: { fontSize: 18, fontWeight: '700', color: Colors.heading },
  optionTextSelected: { color: Colors.primaryAction },
  primaryButton: {
    backgroundColor: Colors.primaryAction, borderRadius: 20,
    paddingVertical: 18, alignItems: 'center',
  },
  primaryText: { fontSize: 20, fontWeight: '700', color: Colors.white },
});
